src/array.cr

```diff
@@ -861,7 +861,7 @@ class Array(T)
   # a.fill(9) # => [9, 9, 9]
   # ```
   def fill(value : T)
-    {% if Int::Primitive.union_types.includes?(T) %}
+    {% if Int::Primitive.union_types.includes?(T) || Float::Primitive.union_types.includes?(T) %}
```

Comment:
Perhaps:

```suggestion
    {% if Number.all_subclasses.includes?(T) %}
```

Reply:
much better. i will fix.

Reply:
NOOOOOOOOOOOOO! Please don't.

This will include BigInt, BigDecimal, etc., and the code will work WRONG.

Reply:
@toddsundsted Please wait until I confirm before making any further changes.

Reply:
i tested before changing, but now i suspect these become subclasses only when required:

https://play.crystal-lang.org/#/r/8ptx

I'll confirm and fix.

Reply:
Yes, subclasses depends on the types you require. Also anyone can subclass `Int` or `Float` so it's pretty bad if we blindly apply this optimization to all numbers.

Reply:
yep, boom.

reverting.

Reply:
thank you!